src/BillWidget.h

```diff
@@ -0,0 +1,135 @@
+     */
+    BillWidget(QObject *parent);
+
+    /**
+     * @brief Destructs the BillWidget by deleting objects which were dynamically allocated.
+     */
+    ~BillWidget();
+
+    /**
+     * @brief Creates the labels, input fields, and buttons of the BillWidget.
+     */
+    void initializeWidget();
+
+    /**
+     * @brief Adds the widgets of the BillWidget into a grid layout.
+     */
+    void createWidgetLayout();
+
+    /**
+     * @brief Returns the close push button pointer.
+     * @return The close push button pointer.
+     */
+    QPushButton* getCloseButton();
+
+    /**
+     * @brief Returns the enter another push button pointer.
+     * @return The enter another push button pointer.
+     */
+    QPushButton* getEnterAnotherButton();
+
+    /**
+     * @brief Returns the done push button pointer.
+     * @return The done push button pointer.
+     */
+    QPushButton* getDoneButton();
+
+    /**
+     * @brief Returns the name line edit pointer.
+     * @return The name line edit pointer.
+     */
+    QLineEdit* getNameInput();
+
+    /**
+     * @brief Returns the amount due line edit pointer.
+     * @return The amount due line edit pointer.
+     */
+    QLineEdit* getAmountDueInput();
+
+    /**
+     * @brief Returns the due date date edit pointer.
+     * @return The due date date edit pointer.
+     */
+    QDateEdit* getDueDateInput();
+
+private:
+
+
+    const QString m_WINDOW_TITLE = "Bill Entry"; //!< Title of the BillWidget window.
+
+    // Name input fields
+    QLabel *m_nameLabel = nullptr; //!< Label for the line edit used for entering the name of the bill.
+    const QString m_NAME_LABEL_TEXT = "Bill Name:"; //!< Text for the label for the line edit used for entering the name of the bill.
+    QLineEdit *m_nameInput = nullptr; //!< The line edit used for entering the name of the bill.
+
+    // Amount due input fields
+    QLabel *m_amountDueLabel = nullptr; //!< Label for the line edit used for entering the amount due of the bill.
+    const QString m_AMOUNT_DUE_LABEL_TEXT = "Amount Due:"; //!< Text for the label for the line edit used for entering the amount due of the bill
+    QLineEdit *m_amountDueInput = nullptr; //!< The line edit used for entering the amount due of the bill.
+
+    // Due date input fields
+    QLabel *m_dueDateLabel; //!< Label for the date edit used for entering the due date of the bill.
+    const QString m_DUE_DATE_LABEL_TEXT = "Due Date:"; //!< Text for the label for the date edit used for entering the due date of the bill.
+    QDateEdit *m_dueDateInput = nullptr; //!< The date edit used for entering the due date of the bill.
+
+    QGridLayout *m_gridLayout = nullptr; //!< Grid layout used for organizing the labels, line edits, date edit, and push buttons of the BillWidget.
+
+    // Button and button text variables
+    QPushButton *m_closeButton = nullptr; //!< Push button used for closing the BillWidget.
```

Comment:
Specify that `m_closeButton` is used for closing the bill widget in order to **exiting the application** and not to display the bill table widget once bill entry is done.